neo/rawio/intanrawio.py

```diff
@@ -183,28 +285,57 @@ def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop, strea
         stream_id = self.header["signal_streams"][stream_index]["id"]
         mask = self.header["signal_channels"]["stream_id"] == stream_id
         signal_channels = self.header["signal_channels"][mask]
+
+        # for channel slicing need to keep track of if None or an actual array that can be indexed
+        channel_indexes_are_none = False
         if channel_indexes is None:
             channel_indexes = slice(None)
+            channel_indexes_are_none = True
         channel_ids = signal_channels["id"][channel_indexes]
 
-        shape = self._raw_data[channel_ids[0]].shape
+        if self.file_format == 'header-attached':
+            shape = self._raw_data[channel_ids[0]].shape
+        elif self.file_format == 'one-file-per-signal':
+            shape = self._raw_data[stream_index].shape
+        else:
+            if channel_indexes_are_none:
+                shape = self._raw_data[stream_index][0].shape
+            else:
+                shape = self._raw_data[stream_index][channel_indexes[0]].shape
 
         # some channel (temperature) have 1D field so shape 1D
         # because 1 sample per block
         if len(shape) == 2:
-            # this is the general case with 2D
+            # this is the general case with 2D (ie in the header-attached format)
             block_size = shape[1]
             block_start = i_start // block_size
             block_stop = i_stop // block_size + 1
 
             sl0 = i_start % block_size
             sl1 = sl0 + (i_stop - i_start)
 
-        sigs_chunk = np.zeros((i_stop - i_start, len(channel_ids)), dtype="uint16")
+        if self.file_format != 'header-attached' and stream_id == 'RHD2000 amplifier channel':
```

Comment:
I think we really need something like what is on:

https://github.com/NeuralEnsemble/python-neo/pull/1416

The time for examples is int32 which is not included here.

Reply:
I did this patch before you submitted your pr. Yours in general is better. No one should request the time. So really the only option was `uint16` but since amplifier is `int16` for the other formats I did mine. I agree yours is the better final solution.

Reply:
Especially since it future proofs in case intan changes other dtypes in the future.

Reply:
How comes the time is not requested? : O

Reply:
Most people just want the data and convert to times themselves by dividing by sample_rate. Also Intan timestamps aren't great if I remember (but I never look at them, except once when they weren't useful for me).

 It's just tricky because I have to do the two different patches so your PR fixes the main case and then I back fix and I clearly get cocky and make mistakes rather than testing locally. I've got the Fridays.... Sorry, but you're right. We should be ready in the case that someone requests time so that we don't give the wrong numbers.

Reply:
Actually I don't believe time is accessible. For something to be accessible for Intan it has to be stored in the ordered_channels. Which if we look in the `read_rhd` and the `read_rhs` functions we add timestamp to the dtype but not to ordered_channels. This was the same problem that was happening with digital data previously before I patched it. Once could reach into the memmap and grab the data, but one couldn't use the machinery.

Reply:
Yes, it is not accessible as of now as you say. I still think that it should be done. But this is also problematic for other datatypes anyway.